### PR TITLE
Update header to match refined mocks. #56.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#56 

@MillenniumFalconMechanic 

### Changes
- modify header to reflect mock refinements

### CODE REVIEW

File: explorer/app/components/Header/components/Logo/logo.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  height?: number;
+  imgSrc: StaticImageData;
+  link: string;
+  width?: number;
+}
+
+export const Logo = ({
+  alt,
+  height,
+  imgSrc,
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Header/components/Logo/logo.tsx:20]: What are your thoughts on updating this to just `src`. The other attributes use a standard HTML attribute naming convention and maybe we can update this one to match?

File: explorer/app/components/Header/components/NavLinks/navLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Core dependencies
+import { Box, Button } from "@mui/material";
+import Link from "next/link";
+import React from "react";
+
+// App dependencies
+import { ELEMENT_ALIGNMENT } from "../../../../common/entities";
+
+export type NavAlignment = Omit<ELEMENT_ALIGNMENT, ELEMENT_ALIGNMENT.RIGHT>;
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Header/components/NavLinks/navLinks.tsx:9]: My apologies, it looks like `Exclude` is the correct utility type to use here! You can use`Omit` is for interfaces and `Exclude` for union types and enums. See [Omit](https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys) and [Exclude](https://www.typescriptlang.org/docs/handbook/utility-types.html#excludeuniontype-excludedmembers) docs.

File: explorer/app/components/Header/components/ProfileComponent/profileComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Core dependencies
+import LoginIcon from "@mui/icons-material/Login";
+import { Button, IconButton, Theme, useMediaQuery } from "@mui/material";
+import React from "react";
+
+export const ProfileComponent = (): JSX.Element => {
+  const desktop = useMediaQuery((theme: Theme) => theme.breakpoints.up("lg"));
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Header/components/ProfileComponent/profileComponent.tsx:7]: Nice!

File: explorer/app/components/StaticImage/StaticImage.tsx [hunk 4]
```diff
@@ -6,21 +6,22 @@
...
 export const StaticImage = ({
+  alt,
+  height,
   src,
   width,
-  height,
-  alt,
 }: StaticImageProps): JSX.Element => {
-  return <img src={src as any} alt={alt} width={width} height={height} />;
+  return <img alt={alt} height={height} src={src as any} width={width} />;
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/StaticImage/StaticImage.tsx:26]: What's the `any` for here?

File: explorer/app/components/common/CustomIcon/customIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+// App dependencies
+import { IconName, IconSvgPathShapes } from "./common/iconSvgPathShapes";
+
+interface Props extends SvgIconProps {
+  iconName: IconName;
+}
+
+export const CustomIcon = ({
+  iconName,
+  ...props /* Spread props to allow for svg icon specific props SvgIconProps e.g. "fontSize", or "htmlColor". */
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/common/CustomIcon/customIcon.tsx:14]: Great comments, thank you!

File: explorer/pages/_app.tsx [hunk 6]
```diff
@@ -1,13 +1,15 @@
...
+import { theme } from "app/theme";
 import { ThemeProvider as Emotion10ThemeProvider } from "@emotion/react";
+import { CssBaseline } from "@mui/material";
 import { ThemeProvider } from "@mui/material/styles";
-import { theme } from "app/theme";
+import type { AppProps } from "next/app";
 
 function MyApp({ Component, pageProps }: AppProps) {
   return (
     <Emotion10ThemeProvider theme={theme}>
```
Comment (MillenniumFalconMechanic) [inline explorer/pages/_app.tsx:10]: Can you remind me why we thought both theme providers are necessary here?

File: explorer/pages/_app.tsx [hunk 7]
```diff
@@ -1,13 +1,15 @@
...
       <ThemeProvider theme={theme}>
+        <CssBaseline />
```
Comment (MillenniumFalconMechanic) [inline explorer/pages/_app.tsx:12]: For my own knowledge, can you tell me what `CssBaseline` is?

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
One nit (Exclude vs Omit!) and a couple of q's! Thanks F.

File: explorer/app/components/Header/components/NavLinks/navLinks.tsx (hunk 2 above)
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Header/components/NavLinks/navLinks.tsx:9]:
```suggestion
export type NavAlignment = Exclude<ELEMENT_ALIGNMENT, ELEMENT_ALIGNMENT.RIGHT>;
```

File: explorer/app/components/StaticImage/StaticImage.tsx (hunk 4 above)
Comment (MillenniumFalconMechanic) [reply explorer/app/components/StaticImage/StaticImage.tsx:26]: Discussed offline; resolve if possible.

File: explorer/pages/_app.tsx (hunk 6 above)
Comment (MillenniumFalconMechanic) [reply explorer/pages/_app.tsx:10]: Let's revisit with [Revisit use of Emotion and MUI theme providers](https://github.com/clevercanary/data-browser/issues/78).

File: explorer/pages/_app.tsx (hunk 7 above)
Comment (MillenniumFalconMechanic) [reply explorer/pages/_app.tsx:12]: Required for global CSS.

File: explorer/app/components/Header/components/Logo/logo.tsx (hunk 1 above)
Comment (frano-m) [reply explorer/app/components/Header/components/Logo/logo.tsx:20]: Done

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Fran!

**Pull request merged**